src/act.movement.cpp

```diff
@@ -679,10 +680,13 @@ void move_vehicle(struct char_data *ch, int dir)
     send_to_char("You aren't the Kool-Aid Man, so you decide against ramming your way out of here.\r\n", ch);
     return;
   }
-  if (!EXIT(veh, dir) || !EXIT(veh, dir)->to_room || EXIT(veh, dir)->to_room == &world[0] || IS_SET(EXIT(veh, dir)->exit_info, EX_CLOSED)) {
+  if (!EXIT(veh, dir) || !EXIT(veh, dir)->to_room || EXIT(veh, dir)->to_room == &world[0] || (IS_SET(EXIT(veh, dir)->exit_info, EX_CLOSED) && (can_enter_closed_apartment == FALSE)) ) {
```

Comment:
I think I led you slightly wrong on this. The code here is checking for crash conditions (the exit must exist, and it must point to a room, and that room must not be limbo), which must be checked before your can_enter_closed_apartment bool is populated. There's also not a need to reuse the check, so we don't actually need the bool (we'd only defined it to make the check more readable and portable, but splitting it up like below makes it readable enough).
```c++
if (!EXIT(veh, dir) 
    || !EXIT(veh, dir)->to_room 
    || EXIT(veh, dir)->to_room == &world[0])
{
    send_to_char(CANNOT_GO_THAT_WAY, ch);
    return;
}

if (IS_SET(EXIT(veh, dir)->exit_info, EX_CLOSED)) {
    if (ROOM_FLAGGED(EXIT(veh, dir)->to_room, ROOM_HOUSE)
        && House_can_enter(ch, EXIT(veh, dir)->to_room->number)
        && has_key(ch, (EXIT(veh, dir)->key)))
    {
        send_to_char("The garage door opener on your key beeps, allowing the door to swing open briefly enough to slide through.\r\n", ch);
        // TODO: print a message to the room about the door auto-opening
    } else {
        send_to_char(CANNOT_GO_THAT_WAY, ch);
        return;
    }
}
```